### PR TITLE
Improve sdk config

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: universal-login-sdk/lib/config/SdkConfig.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import {Message, ObservedToken} from '@universal-login/commons';
+import {PartialRequired, SignedMessage, ObservedToken} from '@universal-login/commons';
 
 export interface SdkConfig {
-  paymentOptions: Message;
+  paymentOptions: PartialRequired<SignedMessage, 'gasPrice' | 'gasLimit' | 'operationType' | 'value' | 'data'>;
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/lib/config/SdkConfig.ts]: Maybe add new type `MessageDefaults`

Comment (marekkirejczyk) [reply universal-login-sdk/lib/config/SdkConfig.ts]: Why not just partial? Why value or data would be required?

File: universal-login-commons/test/core/utils/deepMerge.ts [hunk 2]
```diff
@@ -41,6 +41,14 @@ describe('UNIT: deepMerge', () => {
     expect(deepMerge(obj1, {})).to.deep.eq(obj1);
   });
 
+  it('non-empty merged to undefined', () => {
+    expect(() => deepMerge(undefined, obj1)).to.throws(`Cannot set property 'a1' of undefined`);
+  });
+
+  it('undefined merged to non-empty', () => {
+    expect(deepMerge(obj1, undefined)).to.deep.eq(obj1);
```
Comment (marekkirejczyk) [inline universal-login-commons/test/core/utils/deepMerge.ts]: 👍

File: universal-login-sdk/lib/config/SdkConfig.ts (hunk 1 above)
Comment (xitronix) [reply universal-login-sdk/lib/config/SdkConfig.ts]: PartialRequired clearly defines what we can put in the config and in the sdk constructor we use DeepPartial which does all fields Partial

File: universal-login-sdk/lib/api/sdk.ts [hunk 3]
```diff
@@ -51,9 +50,7 @@ class UniversalLoginSDK {
     this.blockchainObserver = new BlockchainObserver(this.blockchainService);
     this.balanceChecker = new BalanceChecker(this.provider);
     this.tokenDetailsService = new TokenDetailsService(this.provider);
-    this.sdkConfig = sdkConfig || SdkConfigDefault;
-    this.sdkConfig.paymentOptions = {...MESSAGE_DEFAULTS, ...SdkConfigDefault.paymentOptions};
-    this.sdkConfig.observedTokens = this.sdkConfig.observedTokens || SdkConfigDefault.observedTokens;
+    this.sdkConfig = deepMerge(SdkConfigDefault, sdkConfig);
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/api/sdk.ts]:
```
classs PaymentOptions {
  gasPrice? : BigNumber,
  gasLimit? : BigNumber
}
this.defaultTranscationOptions = deepMerge(MESSAGE_DEFAULTS, SdkConfigDefault.paymentOptions)
```

File: universal-login-sdk/lib/api/sdk.ts [hunk 4]
```diff
@@ -39,7 +38,7 @@ class UniversalLoginSDK {
   constructor(
     relayerUrl: string,
     providerOrUrl: string | providers.Provider,
-    sdkConfig?: SdkConfig
+    sdkConfig?: DeepPartial<SdkConfig>
```
Comment (marekkirejczyk) [inline universal-login-sdk/lib/api/sdk.ts]:
```ts
...
sdkConfig?: DeepPartial<SdkConfig>  = SdkConfigDefault
```

**Pull request merged**